Whiskey/Functions/Install-WhiskeyDotNetTool.ps1

```diff
@@ -42,93 +43,114 @@ function Install-WhiskeyDotNetTool
-        if ($globalJsonRollForward -notin $validRollForwardValues)
+        $globalJsonPath = Join-Path -Path $WorkingDirectory -ChildPath 'global.json'
+        if ( -not (Test-Path -Path $globalJsonPath -PathType Leaf) )
         {
-            $msg = "sdk.rollForward value ""$($globalJsonRollForward)"" in ""$($globalJsonPath)"" is not " +
-            "one of the valid .NET SDK roll forward strategies: $($validRollForwardValues -join ', ')." +
-            "$(([Environment]::NewLine)*2)If ""$($globalJsonRollForward)"" is a new valid .NET SDK roll forward strategy " +
-            "then submit a Whiskey issue (https://github.com/webmd-health-services/Whiskey/issues) asking us to add " +
-            "support for it.$(([Environment]::NewLine)*2)Defaulting roll forward strategy to ""$($rollForwardStrategy)""."
-            Write-Warning -Message $msg
+            $globalJsonPath = Join-Path -Path $InstallRoot -ChildPath 'global.json'
         }
-        else
+
+        $sdkVersion = $null
+        if ( Test-Path -Path $globalJsonPath -PathType Leaf )
         {
-            $rollForwardStrategy = [Whiskey.DotNetSdkRollForward] $globalJsonRollForward
-        }
+            try
+            {
+                $globalJson = Get-Content -Path $globalJsonPath -Raw | ConvertFrom-Json
+            }
+            catch
+            {
+                $msg = "Failed to install .NET because global.json file ""$($globalJsonPath)"" contains invalid JSON."
+                Write-WhiskeyError -Message $msg
+                return
+            }
+
+            $globalJsonSdkOptions =
+                $globalJson |
+                Select-Object -ExpandProperty 'sdk' -ErrorAction Ignore
+
+            $globalJsonVersion =
+                $globalJsonSdkOptions |
+                Select-Object -ExpandProperty 'version' -ErrorAction Ignore
+
+            $globalJsonRollForward =
+                $globalJsonSdkOptions |
+                Select-Object -ExpandProperty 'rollForward' -ErrorAction Ignore
+
+            $rollForwardStrategy = [WhiskeyDotNetSdkRollForward]::Disable
+            $validRollForwardValues = [WhiskeyDotNetSdkRollForward].GetEnumNames()
+
+            if ($globalJsonRollForward)
+            {
+                if ($globalJsonRollForward -and $globalJsonRollForward -notin $validRollForwardValues)
+                {
+                    $msg = "Using default roll forward strategy ""$($rollForwardStrategy)"" because the " +
+                           "sdk.rollForward value ""$($globalJsonRollForward)"" in ""$($globalJsonPath)"" is not " +
+                           "recognized. Accepted values are: $($validRollForwardValues -join ', ')." +
+                           [Environment]::NewLine + [Environment]::NewLine +
+                           "If ""$($globalJsonRollForward)"" is a new roll forward strategy, " +
+                           '[submit a Whiskey issue](https://github.com/webmd-health-services/Whiskey/issues) ' +

```

Comment:
Markdown formatting in the console?

Reply:
Yes? We use it in our docs. Should we not?

Reply:
We should make console/log output as easy to read as possible and un-rendered Markdown isn't always the easiest to read, especially if someone is unfamiliar with it. With our docs the hope is one day we can extract them and render them in Markdown somewhere, not-so with console output. So my vote is console output should just be easy-to-read plain text.